### PR TITLE
CNF-11977: e2e: Added mixed cpus negative tests

### PR DESCRIPTION
**Pull request by rbaturov**
Added the following tests:
* A best-effort pod effort can't be deployed with shared-cpus resource
* A burstable pod can't be deployed with shared-cpus resource
* A pod with shared-cpus resource can't be deployed in ns without mixedcpus annotation.
* A pod with an invalid resource openshift.io/enable-shared-cpus != 1, should fail.

### CODE REVIEW

**Comment by rbaturov** [comment]
/hold
potentially will add more tests

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 1]
```diff
@@ -320,6 +320,77 @@ var _ = Describe("Mixedcpus", Ordered, func() {
 				// in the child processes
 			})
 		})
+		When("trying to deploy a burstable pod", func() {
+			It("should cause the burstable pod to fail", func() {
+
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:325]: please redundant empty lines

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 2]
```diff
@@ -320,6 +320,77 @@ var _ = Describe("Mixedcpus", Ordered, func() {
...
+				By("Trying to create a burstable pod")
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:326]: creating a burstable pod

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 3]
```diff
@@ -320,6 +320,77 @@ var _ = Describe("Mixedcpus", Ordered, func() {
...
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				Expect(err.Error()).To(ContainSubstring("resource but pod is not Guaranteed QoS class"))
+			})
+		})
+		When("trying to deploy a best effort pod", func() {
+			It("should cause the best effort pod to fail", func() {
+
+				By("Trying to create a best effort pod")
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:341]: creating best effort pod

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 4]
```diff
@@ -320,6 +320,77 @@ var _ = Describe("Mixedcpus", Ordered, func() {
...
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				Expect(err.Error()).To(ContainSubstring("resource but pod is not Guaranteed QoS class"))
+			})
+		})
+		When("creating a pod with resource openshift.io/enable-shared-cpus != 1", func() {
+			It("should cause the pod to fail and not created", func() {
+
+				By("Trying to create a pod with an incompatible resource request")
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:355]: create a pod with an incompatible resource request

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 5]
```diff
@@ -320,6 +320,77 @@ var _ = Describe("Mixedcpus", Ordered, func() {
...
+				_, err := createPod(ctx, testclient.Client, testutils.NamespaceTesting,
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				expectedError := "more than a single \"workload.openshift.io/enable-shared-cpus\" resource is forbiden, please set the request to 1 or remove it"
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(expectedError))
+			})
+		})
+		When("using namespace with mixedcpus disabled", func() {
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:370]: using namespace without mixedcpus allowed annotation

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 6]
```diff
@@ -320,6 +320,77 @@ var _ = Describe("Mixedcpus", Ordered, func() {
...
+			It("should not be able to deploy a pod requesting shared cpus", func() {
+
+				By("Updating the namespace to disable mixedcpus")
```
Comment (Tal-or) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:373]: keep lower case over all `By` for consistency

**Review by Tal-or — COMMENTED** [review]
Good work, nit comments

**Comment by rbaturov** [comment]
> Good work, nit comments

Thank you!
Will update the code according to these comments.

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 7]
```diff
@@ -320,6 +320,73 @@ var _ = Describe("Mixedcpus", Ordered, func() {
...
+				By("creating a burstable pod")
+				rl := &corev1.ResourceList{
+					corev1.ResourceCPU: resource.MustParse("1"),
+					sharedCpusResource: resource.MustParse("1"),
+				}
+				_, err := createPod(ctx, testclient.Client, testutils.NamespaceTesting,
+					withRequests(rl),
+					withLimits(rl),
+					withRuntime(components.GetComponentName(profile.Name, components.ComponentNamePrefix)))
+				Expect(err.Error()).To(ContainSubstring("resource but pod is not Guaranteed QoS class"))
```
Comment (ffromani) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:334]: this (substring in error message) is an antipattern but I don't have a quick fix handy so let's treat like technical debt and fix "later" ( :\ ).

File: test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go [hunk 8]
```diff
@@ -320,6 +320,73 @@ var _ = Describe("Mixedcpus", Ordered, func() {
 				// in the child processes
 			})
 		})
+		When("trying to deploy a burstable pod", func() {
+			It("should cause the burstable pod to fail", func() {
+				By("creating a burstable pod")
```
Comment (ffromani) [inline test/e2e/performanceprofile/functests/11_mixedcpus/mixedcpus.go:325]: if you have just a `By` you can omit them

**Review by ffromani — COMMENTED** [review]
/approve

we want these tests

**Pull request merged**